extensions/data-plane/data-plane-azure-storage/src/test/java/org/eclipse/edc/connector/dataplane/azure/storage/metadata/BlobMetadataProviderImplTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) 2023 Robert Bosch GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Robert Bosch GmbH - initial implementation
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.azure.storage.metadata;
+
+import org.eclipse.edc.connector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.types.domain.transfer.DataFlowRequest;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+
+public class BlobMetadataProviderImplTest {
+
+    @Test
+    public void multiple_succeeds() {
+
+        var monitor = mock(Monitor.class);
+        var provider = new BlobMetadataProviderImpl(monitor);
+        var requestMock = mock(DataFlowRequest.class);
```

Comment:
pls do not mock value objects. either use a dummy (private nested class), or an actual implementation, e.g. `AzureStoragePart`.